### DISCUSSION TITLE
Creation and Submission of Icons - Contribution, Design, Cleanup, and more

### DISCUSSION THREAD

**Discussion by optimizasean** [tailwindlabs/heroicons#611] [General]
I see almost all issues here are open asking for different icons.  It is a good problem to have the demand but due to the amount of work it takes to create them, would it be possible to submit icons to you for tweaking and approval to bring them into the feature set Heroicons provides?  I think including something like this would be a good way to at least save you time developing icons as you can either bring in the good ones (with or without tweaks) or use them as an inspiration to make your own and therefore save time on design.  It would also help those of us who use Heroicons in our projects to get more icons and potentially case specific icons faster.

Potential Requirements:

1. Contribution document - How to design (keep stroke, fill, view box, size, style consistent), what program to use and how to optimize.
2. Directions for submission and review as well as how to generate the jsx and put on Heroicons website
3. Templates for designing - both outline and solid already set to the proper bounding box and size and other settings
4. Project board to prioritize which icons should be made and maintain a running log of icons to do and who is tackling them to avoid rework (this would also thin out your issue section and help close out all non fix/feature issues as they are added to the board to ensure issues don't get drowned in icon requests and lose important issues in the flood).

I wouldn't want to end up with icons that don't match the style or that don't fit the set standards of size/color/stroke/fill and so on either since then the set would not be as nice as it is now which is why I prefer this to all other sets (also because of the optimization and compatibility with TailwindCSS).

I assume there are a number of designers and others who use these icons and would love to help out (I'm not, just a programmer but wouldn't mind attempting to submit some stuff but might be able to help out).  Thoughts?

**Comment by optimizasean** [comment]
I see some great submissions such as this issue: #130 

There are also a number of pull requests adding and cleaning things but there is no standardization or guide on how you would like this.  I would like to see details on how the community could best support this initiative in a way that works with your workflow and preserves the quality of Heroicons.

**Comment by simonswiss** [comment]
Hey!

Thanks for these thoughts and sorry for the lack of response! Going to convert this to a discussion instead of an issue, perhaps a better place to give this some engagement.

I cannot guarantee that community-contributed icons will be something that makes the Heroicons collection, but this is certainly an interesting point of discussion 👍

**Reply by optimizasean** [reply]
Thanks!
